haystack/preview/components/file_converters/markdown.py

```diff
@@ -0,0 +1,119 @@
+import logging
+import re
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union, Any
+
+from tqdm import tqdm
+
+from haystack.preview import Document, component
+from haystack.preview.lazy_imports import LazyImport
+
+with LazyImport("Run 'pip install beautifulsoup4 markdown python-frontmatter'") as markdown_conversion_imports:
+    import frontmatter
+    from bs4 import BeautifulSoup, NavigableString
+    from markdown import markdown
+
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class MarkdownToTextDocument:
+    """
+    Converts a Markdown file into a text Document.
```

Comment:
Please also add a simple usage example, as we have recently begun to do.

An example: https://github.com/deepset-ai/haystack/blob/3ff79d0307db167ce128ac235ac6e5215607dea6/haystack/preview/components/generators/hugging_face/hugging_face_local.py#L70